README.md

```diff
@@ -22,25 +24,126 @@ pip install framegrab
 
 ### Frame Grabbing
 
-To start grabbing frames from a camera or stream, first create a `FrameGrabber` object, specifying the desired stream source and target FPS:
+Simple usage with a single webcam would look something like the following:
 
 ```
 from framegrab import FrameGrabber
 
-STREAM = 'your_stream_here'
-FPS = 5
+config = {
+    'input_type': 'webcam',
+}
 
-grabber = FrameGrabber.create_grabber(stream=STREAM, fps_target=FPS)
-```
+grabber = FrameGrabber.create_grabber(config)
 
-Replace `'your_stream_here'` with the desired stream source. The `STREAM` variable can be an integer representing the device number, a filename, or a URL of a video stream (e.g., `rtsp://host:port/script?params`, `movie.mp4`, or `*.jpg`).
+```
+`config` can contain many details and settings about your camera, but the only required parameter is `input_type`. Available `input_types` are: `webcam`, `rtsp`, `realsense` and `basler_usb`.
 
-To grab a frame, simply call the `grab()` method:
+Using a single webcam with several options would look something like this:
+```
+config = {
+    'name': 'front door camera',
+    'input_type': 'webcam',
+    'id': {
+        'serial_number': 23432570
+    },
+    'options': {
+        'resolution': {
+            'height': 1080,
+            'width': 1920,
+        },
+        'zoom': {
+            'digital': 1.5
+        }
+
+    }
+}
+
+grabber = FrameGrabber.create_grabber(config)
+```
 
+To get a frame, simply run:
 ```
 frame = grabber.grab()
-if frame is None:
-    print("No frame captured!")
+```
+You can also change the options after the grabber is created.
+```
+new_options = {
+    'resolution': {
+        'height': 480,
+        'width': 640,
+    },
+    'crop': {
+        'relative': {
+            'top': .1,
+            'bottom': .9,
+            'left': .1,
+            'right': .9,
+        }
+    }
+}
+
+grabber.apply_options(new_options)
+```
+
+When you are done with the camera, release the resource by running:
+```
+grabber.release()
+```
+
+You might have several cameras of different types that you want to use in the same application. In this case, you can load the configurations from a yaml file and use `FrameGrabber.create_grabbers`.
+
+If you have multiple cameras of the same type plugged in, it's recommended to provide serial numbers in your configurations; this will ensure that each configuration is paired with the correct camera. If you don't provide serial numbers in your configurations, configurations will be paired with cameras in a sequential manner. 
+
+Below is a sample yaml file containing configurations for three different cameras.
+```
+GL_CAMERAS: |
+  - name: on robot arm
+    input_type: realsense
+    options: 
+      depth:
+        side_by_side: 1
+      crop:
+        relative:
+          right: .8
+  - name: conference room
+      input_type: rtsp
+      address: 
+        rtsp_url: <YOUR RTSP URL>
```

Comment:
```suggestion
        rtsp_url: rtsp://admin:password@192.168.1.20/stream1
```

An actual example makes the docs more helpful.

Reply:
version number updated and RTSP URL corrected